jbehave-support-core/src/main/java/org/jbehavesupport/core/internal/web/WebScreenshotCreator.java

```diff
@@ -25,24 +25,34 @@
 @RequiredArgsConstructor
 public class WebScreenshotCreator {
 
+    public enum type {FAILED, DEBUG, MANUAL, WAIT, STEP}
+
     public static final String FAILED_SCREENSHOTS_KEY = "error_screenshots";
     public static final String STEP_SCREENSHOTS_KEY = "step_screenshots";
-    public static final String FAILED = "FAILED";
-    public static final String STEP_SCREENSHOT = "STEP_SCREENSHOT";
     private static final String FILE_NAME_PATTERN = "%s_%s.png";
 
+    @Value("${web.screenshot.type:MANUAL}")
+    private type desiredType;
+
     @Value("${web.screenshot.directory:./target/reports}")
     private String screenshotDirectory;
-    private String screenShotType;
+    private type screenShotType;
 
     private final WebDriver driver;
     private final TestContext testContext;
 
-    public final void createScreenshot(String screenShotType) {
+    public final void createScreenshot(type screenShotType) {
         this.screenShotType = screenShotType;
+        if (screenShotType == type.FAILED || screenShotType == desiredType || screenShotType == type.MANUAL ||
+            (screenShotType == type.STEP && desiredType == type.DEBUG)) {
+            createScreenshot();
+        }
+    }
+
+    private void createScreenshot() {
         try {
             if (driver instanceof TakesScreenshot) {
-                log.info("Taking {} screenshot will place it in {}", screenShotType.equals(FAILED) ? "error" : "step", screenshotDirectory);
+                log.info("Taking {} screenshot will place it in {}", screenShotType.equals(type.FAILED) ? "error" : "step", screenshotDirectory);
```

Comment:
How about just:
`log.info("Taking {} screenshot will place it in {}", screenShotType, screenshotDirectory);`

Reply:
Good idea